### PR TITLE
:sparkles: added pressboard as an analytics vendor

### PR DESCRIPTION
**Pull request by omar84**
Adding Pressboard as an analytics vendor to amp-analytics

### CODE REVIEW

**Comment by omar84** [comment]
I signed it!

**Comment by omar84** [comment]
Hi,

I fixed a bunch of bugs, now LGTM is passing, but Travis is failing on stuff that is not related to my PR.
I'm not sure what is wrong with percy, and my manager signed the CLA. not sure what to do now.

**Comment by omar84** [comment]
@zhouyx can you please take a look? all checks have passed successfully. thanks!

**Comment by zhouyx** [comment]
Thank you @omar84. I'll take a look soon. One early comment could you also add your example to this [file](https://github.com/ampproject/amphtml/blob/master/examples/analytics-vendors.amp.html). So it will be easy for us to test your change. Thank you.

**Comment by omar84** [comment]
Thanks for the note @zhouyx , I added an example. the config endpoint is not live yet though, is that ok? or do I need it to be live so you can see the data returned?

when it will become live tomorrow, its output will look like this: 

`{  
   "vars":{  
      "geoNameId":"6173331",
      "country":"Canada",
      "region":"British Columbia",
      "city":"Vancouver",
      "mediaId":"5",
      "campaignId":"10041",
      "storyRequestId":"1171",
      "visitorId":"432b8bb2-9b6d-4d1a-9e12-9afc6981be78",
      "dbInstance":"pbr3",
      "timeZoneOffset":"-8"
   }
}`

File: extensions/amp-analytics/0.1/vendors/pressboard.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        },
+        'stopSpec': {
+          'on': 'hidden',
+        },
+      },
+      'request': 'conversion',
+    },
+  },
+  'transport': {
+    'beacon': false,
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/vendors/pressboard.js]: To confirm that your server doesn't support `xhrpost` or `beacon`? Asking because it is recommended to use those two transport method, and AMP will handle the fallback case if there's no browser support.

File: extensions/amp-analytics/0.1/vendors/pressboard.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      'on': 'visible',
+      'request': 'conversion',
+    },
+    'trackAnchorClicks': {
+      'on': 'click',
+      'selector': 'a',
+      'request': 'conversion',
+    },
+    'pageTimer': {
+      'on': 'timer',
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/vendors/pressboard.js]: I need to double check on the timer trigger. Once it is checked in here, it will be included as a default trigger. Asking because this can greatly increase the number of requests sent to your server? Is that ok? And you want the timer trigger by default? Thanks

File: extensions/amp-analytics/0.1/vendors/pressboard.js (hunk 1 above)
Comment (omar84) [reply extensions/amp-analytics/0.1/vendors/pressboard.js]: confirmed, we don't have support for either for now.

**Comment by zhouyx** [comment]
One more comment
https://github.com/ampproject/amphtml/pull/18169#discussion_r219598759

File: extensions/amp-analytics/0.1/vendors/pressboard.js (hunk 2 above)
Comment (omar84) [reply extensions/amp-analytics/0.1/vendors/pressboard.js]: can we turn it off based on an input coming back from config?

**Comment by omar84** [comment]
@zhouyx , I replied :), any chance a merge is coming soon?

File: extensions/amp-analytics/0.1/vendors/pressboard.js (hunk 2 above)
Comment (zhouyx) [reply extensions/amp-analytics/0.1/vendors/pressboard.js]: I see. Can I know what the override config looks like? 

Another question I have is:  the timer trigger is the only trigger you have here.  If it is disabled, does that mean there will be no trigger at all?

Comment (omar84) [reply extensions/amp-analytics/0.1/vendors/pressboard.js]: yes, for now all we do is send back user attention variables for our stories, if its not our story, then we don't send back anything

our override for stories that are not ours looks like this: 

{
        vars: {
            mediaId: someMediaId,
        },
        triggers: {
            pageTimer: {
                on: 'never',
            },
        }
    }

**Comment by omar84** [comment]
@zhouyx I replied again, I hope everything is good to go.

**Comment by omar84** [comment]
@zhouyx , did you get a chance to check my comments?
sorry if I'm being annoying, but we can't wait to make this live.

**Comment by omar84** [comment]
@zhouyx , any update?

File: extensions/amp-analytics/0.1/vendors/pressboard.js (hunk 2 above)
Comment (omar84) [reply extensions/amp-analytics/0.1/vendors/pressboard.js]: @zhouyx , any update?

**Comment by omar84** [comment]
@zhouyx any update?

**Comment by omar84** [comment]
@zhouyx , @choumx guys please, any update?

**Comment by choumx** [comment]
Sorry for the delay, this week was the [AMP Contributor Summit](https://events.withgoogle.com/amp-contributor-summit/) so most of the team has been busy.

File: extensions/amp-analytics/0.1/vendors/pressboard.js (hunk 2 above)
Comment (zhouyx) [reply extensions/amp-analytics/0.1/vendors/pressboard.js]: @omar84 even with `on: 'never'` a trigger that wait for the custom event `never` will be created. To avoid the waste there, you can use the `enabled` config that we support to make disable the pageTimer trigger. Note `enabled` doesn't support boolean right now, I'll send out a quick PR to fix it. 

Or you can return the `pageTimer` trigger along with the remote config. 

enabled config looks like
```
triggers: {
      pageTimer: {
           enabled: false,
       },
}
```

**Comment by zhouyx** [comment]
@omar84 Sorry for the delay. Was in the AMP contributor summit. I left one comment regarding disabling the trigger.

File: extensions/amp-analytics/0.1/vendors/pressboard.js (hunk 2 above)
Comment (omar84) [reply extensions/amp-analytics/0.1/vendors/pressboard.js]: done, I updated the code in our server to return enabled: false instead of on: 'never'

its not live yet, but will be soon.

**Comment by omar84** [comment]
@zhouyx thanks, sorry for my nagging :), updated the trigger config as you suggested

**Comment by omar84** [comment]
@zhouyx and I added pressboard as an amp-ad vendor too

**Comment by zhouyx** [comment]
@omar84 LGTM to the analytics related change. Please separate the ad related change to a different PR. It's easier for review.

**Comment by omar84** [comment]
@zhouyx I pulled the latest, merged my pull request, and all seems good, can we merge?

**Comment by zhouyx** [comment]
@omar84 The analytics related change looks good. But please separate the ad related change to a different PR. https://github.com/ampproject/amphtml/pull/18169#issuecomment-425602076
Thanks!

File: examples/ads.amp.html [hunk 3]
```diff
@@ -181,6 +180,7 @@
         <option>pubmine</option>
         <option>pulsepoint</option>
         <option>purch</option>
+        <option>capirs</option>
```
Comment (zhouyx) [inline examples/ads.amp.html]: oops please revert the change here. Names are sorted by alphabetical order.

Comment (omar84) [reply examples/ads.amp.html]: done

**Comment by omar84** [comment]
@zhouyx done, all good now.

**Comment by omar84** [comment]
Thank you @zhouyx !

**Comment by zhouyx** [comment]
Thank you @omar84. Please follow up with a PR to the ampproject/docs repo to add your vendor name to the supported vendor list. And your ad related code in another PR : )

**Comment by omar84** [comment]
@zhouyx , done. should I close the intent to implement issue that referenced this pull request?
